spec/requests/csp_violation_report_spec.rb

```diff
@@ -0,0 +1,25 @@
+require 'rails_helper'
+
+describe 'csp violations', type: :request do
+  it 'records the violation' do
+    body = {
+      "csp-report" => {
+        "blocked-uri" => "data",
+        "document-uri" => "http://localhost:3000/s/izi825/hckr_news_hacker_news_sorted_by_time",
+        "original-policy" => [
+          "default-src 'none'",
+          "img-src *",
+          "script-src 'self' 'unsafe-inline'",
+          "style-src 'self' 'unsafe-inline'",
+          "form-action 'self'",
+          "report-uri http://localhost:3000/csp-violation-report",
+        ].join('; '),
+        "referrer" => "http://localhost:3000/",
+        "violated-directive" => "img-src",
+      },
+    }
+
+    post '/csp-violation-report', params: { body: body.to_json, format: :json }
+    expect(response).to have_http_status(:ok)
```

Comment:
Although this is strictly better than no test at all, this isn't really testing that it records the violation, so the it-message is somewhat misleading. How about changing the it-message to "responds appropriately" or adding an expectation that ensures the violation is being logged, maybe like this:
```ruby
it 'records the violation' do
  allow(Rails.logger).to receive(:info).with(expected_body)
  post '/csp-violation-report', params: { body: actual_body.to_json, format: :json }
  expect(Rails.logger).to have_received(:info).with(expected_body)
end
```
`expected_body` and `actual_body` could be set in let-blocks above.

Reply:
LGTM 👍